### PR TITLE
Feature Uribuilder

### PR DESCRIPTION
**Pull request by linawolf**
Updated examples for edit or create record links. There is now a viewhelper
in the sysextension "backend" therefore the custom viewhelper could be
removed.

### CODE REVIEW

File: Classes/Controller/ModuleController.php [hunk 1]
```diff
@@ -281,7 +222,54 @@ public function clipboardAction()
      */
     public function linksAction()
     {
+        $backendUriBuilder = GeneralUtility::makeInstance(\TYPO3\CMS\Backend\Routing\UriBuilder::class);
```
Comment (mbrodala) [inline Classes/Controller/ModuleController.php:225]: We should `use` this. `as BackendUriBuilder` if necessary.

File: Classes/Controller/ModuleController.php [hunk 2]
```diff
@@ -336,6 +324,16 @@ public function fileReferenceAction($element = 0)
...
     }
 
+    /**
+     * Returns the global database connection object.
+     *
+     * @return \TYPO3\CMS\Core\Database\DatabaseConnection
+     */
+    protected function getDatabaseConnection()
+    {
+        return $GLOBALS['TYPO3_DB'];
```
Comment (mbrodala) [inline Classes/Controller/ModuleController.php:334]: Uh no, not anymore. ;-)

File: Configuration/TCA/tx_examples_haiku.php [hunk 3]
```diff
@@ -157,6 +164,7 @@
                 ],
             ],
         ],
+        /*
```
Comment (mbrodala) [inline Configuration/TCA/tx_examples_haiku.php:167]: Hm, why not drop instead? Also is this related to backend `UriBuilder` or just a general update?

**Review by mbrodala — CHANGES_REQUESTED** [review]
There seems to be a general mix with #8 Did you miss a rebase?

File: Configuration/TCA/tx_examples_haiku.php (hunk 3 above)
Comment (linawolf) [reply Configuration/TCA/tx_examples_haiku.php:167]: The problems in the haiku TCA throwed errors when trying to call the corresponding backend link functions so I fixed them here

File: Classes/Controller/ModuleController.php (hunk 2 above)
Comment (linawolf) [reply Classes/Controller/ModuleController.php:334]: strange, actually I removed the part. stange that it looks as if I added it...

File: Classes/Controller/ModuleController.php (hunk 1 above)
Comment (linawolf) [reply Classes/Controller/ModuleController.php:225]: what do you mean? Is there a class called "BackendUriBuilder"?

Comment (mbrodala) [reply Classes/Controller/ModuleController.php:225]: Yes, if you name it accordingly:

```
use TYPO3\CMS\Backend\Routing\UriBuilder as BackendUriBuilder;
```

This way you avoid conflicts and confusion with the Extbase `UriBuilder` (which you could also rename to `ExtbaseUriBuilder` instead but that'd be rather unusual.)

File: Configuration/TCA/tx_examples_haiku.php (hunk 3 above)
Comment (mbrodala) [reply Configuration/TCA/tx_examples_haiku.php:167]: OK, but then we can probably drop this altogether?

Comment (linawolf) [reply Configuration/TCA/tx_examples_haiku.php]: ok, did so just now

File: Classes/Controller/ModuleController.php (hunk 2 above)
Comment (linawolf) [reply Classes/Controller/ModuleController.php:335]: it is removed

File: Classes/Controller/ModuleController.php (hunk 1 above)
Comment (linawolf) [reply Classes/Controller/ModuleController.php]: Ok done

File: Configuration/TCA/tx_examples_haiku.php (hunk 3 above)
Comment (linawolf) [reply Configuration/TCA/tx_examples_haiku.php]: done

File: Classes/Controller/ModuleController.php (hunk 2 above)
Comment (mbrodala) [reply Classes/Controller/ModuleController.php:335]: This still is marked as "added" ... :thinking:

Comment (linawolf) [reply Classes/Controller/ModuleController.php:335]: I remove it in another pull request. I split up the changes into chunks of one topic. the uribuiler doesnt touch the db things

**Pull request merged**